packages/core/src/modules/mdoc/__tests__/mdoc.deviceResponse.openid4vp.test.ts

```diff
@@ -0,0 +1,305 @@
+            issuing_country: 'US',
+            issuing_authority: 'NY DMV',
+            document_number: '01-856-5050',
+            portrait: 'bstr',
+            driving_privileges: [
+              {
+                vehicle_category_code: 'C',
+                issue_date: '2023-09-01',
+                expiry_date: '2028-09-31',
+              },
+            ],
+            un_distinguishing_sign: 'tbd-us.ny.dmv',
+
+            sex: 'F',
+            height: '5\' 8"',
+            weight: '120lb',
+            eye_colour: 'brown',
+            hair_colour: 'brown',
+            resident_addres: '123 Street Rd',
+            resident_city: 'Brooklyn',
+            resident_state: 'NY',
+            resident_postal_code: '19001',
+            resident_country: 'US',
+            issuing_jurisdiction: 'New York',
+          },
+        },
+      })
+    }
+
+    //  This is the Device side
+    {
+      const result = await MdocDeviceResponse.openId4Vp(agent.context, {
+        mdocs: [mdoc],
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        presentationDefinition: PRESENTATION_DEFINITION_1 as any,
+        sessionTranscriptOptions: {
+          clientId,
+          responseUri,
+          verifierGeneratedNonce,
+          mdocGeneratedNonce,
+        },
+        deviceNameSpaces: {
+          'com.foobar-device': { test: 1234 },
+        },
+      })
+      deviceResponse = result.deviceResponseBase64Url
+
+      const parsed = parseDeviceResponse(TypedArrayEncoder.fromBase64(deviceResponse))
+      expect(parsed.documents).toHaveLength(1)
+      parsedDocument = Mdoc._interalFromIssuerSignedDocument(parsed.documents[0])
+    }
+  })
+
+  it('should be verifiable', async () => {
+    const res = await MdocDeviceResponse.verify(agent.context, {
+      deviceResponse,
+      trustedCertificates: [ISSUER_CERTIFICATE],
+      mdoc,
+      sessionTranscriptOptions: {
+        clientId,
+        responseUri,
+        verifierGeneratedNonce,
+        mdocGeneratedNonce,
+      },
+    })
+    expect(res).toHaveLength(1)
+  })
+
+  describe('should not be verifiable', () => {
+    ;[
+      [
+        'clientId',
+        {
+          clientId: 'wrong',
+          responseUri,
+          verifierGeneratedNonce,
+          mdocGeneratedNonce,
+        },
+      ] as const,
+      [
+        'responseUri',
+        {
+          clientId,
+          responseUri: 'wrong',
+          verifierGeneratedNonce,
+          mdocGeneratedNonce,
+        },
+      ] as const,
+      [
+        'verifierGeneratedNonce',
+        {
+          clientId,
+          responseUri,
+          verifierGeneratedNonce: 'wrong',
+          mdocGeneratedNonce,
+        },
+      ] as const,
+      [
+        'mdocGeneratedNonce',
+        {
+          clientId,
+          responseUri,
+          verifierGeneratedNonce,
+          mdocGeneratedNonce: 'wrong',
+        },
+      ] as const,
+    ].forEach(([name, values]) => {
```

Comment:
I think this might look a bit clear?

```suggestion
    Object.entries({
      clientId: {
          clientId: 'wrong',
          responseUri,
          verifierGeneratedNonce,
          mdocGeneratedNonce,
        } 
        }).forEach(([name, values]) => {
```